src/transactions/TransactionFrame.cpp

```diff
@@ -740,6 +740,19 @@ TransactionFrame::applyOperations(SignatureChecker& signatureChecker,
                                   TransactionMeta& outerMeta)
 {
     ZoneScoped;
+    auto& internalErrorCounter = app.getMetrics().NewCounter(
+        {"ledger", "transaction", "internal-error"});
+
+    // We do not want to treat exception as internal-error for non-live ledger
```

Comment:
Nitpick 1: this comment is a bit confusing. We do want to treat exceptions as internal errors, but we don't want to increment the metric internal-error. Maybe more precise language would help here.

Nitpick 2: we normally say "ledger header" instead of "head of the ledger"